cdc/kv/shared_stream.go

```diff
@@ -90,12 +90,28 @@ func newStream(ctx context.Context, c *SharedClient, g *errgroup.Group, r *reque
 			if err := waitForPreFetching(); err != nil {
 				return err
 			}
-			if canceled := stream.run(ctx, c, r); canceled {
-				return nil
+			var regionErr error
+			if err := version.CheckStoreVersion(ctx, c.pd, r.storeID); err != nil {
+				log.Info("event feed check store version fails",
+					zap.String("namespace", c.changefeed.Namespace),
+					zap.String("changefeed", c.changefeed.ID),
+					zap.Uint64("streamID", stream.streamID),
+					zap.Uint64("storeID", r.storeID),
+					zap.String("addr", r.storeAddr),
+					zap.Error(err))
+				if errors.Cause(err) == context.Canceled {
+					return nil
+				}
+				regionErr = &getStoreErr{}
```

Comment:
It's better to move `getStoreErr` into `CheckStoreVersion`, because `CheckStoreVersion` may fails really without the `getStore` bug.

Reply:
OK, change it to check the error type return by `CheckStoreVersion`.